### PR TITLE
Add unit tests to TemplateDataUtil class

### PR DESCRIPTION
**Pull request by wuyuxin511**

### CODE REVIEW

File: src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "{'mysql_version':'5.6','password':'dfhieuwhfhdsfj','wordpress_version':'4.8-apache'}";
+
+    Map<String, String> resultMap = TemplateDataUtil.parseTemplateDataJson(templateDataJson);
+
+    Map<String, String> expectedMap = new HashMap<>();
+    expectedMap.put("mysql_version", "5.6");
+    expectedMap.put("password", "dfhieuwhfhdsfj");
+    expectedMap.put("wordpress_version", "4.8-apache");
+
+    assertThat(resultMap).isEqualTo(expectedMap);
```
Comment (magl0) [inline src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java]: nit: `assertThat(...).containsExactly(...)`.

File: src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+}
```
Comment (magl0) [inline src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java:74]: Add at least 3 more test cases: parsing empty map data `"{}"`, parsing empty string, and parsing non-map json string.

File: src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        "{'mysql_version':'5.6','password':'dfhieuwhfhdsfj','wordpress_version':'4.8-apache'}";
+
+    Map<String, String> resultMap = TemplateDataUtil.parseTemplateDataJson(templateDataJson);
+
+    Map<String, String> expectedMap = new HashMap<>();
+    expectedMap.put("mysql_version", "5.6");
+    expectedMap.put("password", "dfhieuwhfhdsfj");
+    expectedMap.put("wordpress_version", "4.8-apache");
+
+    assertThat(resultMap).containsExactlyEntriesIn(expectedMap);
```
Comment (magl0) [inline src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java]: nitpick: `assertThat(resultMap).containsExactly("mysql_version", "5.6", "password", ...);`

File: src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  @Test
+  public void parseTemplateData_whenMapDataEmpty_success() {
+    String templateDataJson = "{}";
+
+    Map<String, String> resultMap = TemplateDataUtil.parseTemplateDataJson(templateDataJson);
+
+    Map<String, String> expectedMap = new HashMap<>();
+
+    assertThat(resultMap).containsExactlyEntriesIn(expectedMap);
```
Comment (magl0) [inline src/test/java/tsunami/security/scanner/utilities/TemplateDataUtilTest.java]: `assertThat(...).isEmpty()`
